### ISSUE TITLE
Unhandled crash when mod found in classpath not in mods.toml

### ISSUE THREAD

**Issue by maxanier** [MinecraftForge/MinecraftForge#6057]
**Minecraft Version:** 1.14.4

**Forge Version:** 28.0.46 (Or rather the forge src from that build)

**Logs:**:
```
java.lang.NullPointerException: Initializing game
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_222] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_222] {}
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_222] {}
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[?:1.8.0_222] {}
	at net.minecraftforge.fml.ModLoader.buildMods(ModLoader.java:215) ~[main/:?] {}
	at net.minecraftforge.fml.ModLoader.lambda$gatherAndInitializeMods$18(ModLoader.java:170) ~[main/:?] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_222] {}
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193) ~[?:1.8.0_222] {}
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472) ~[?:1.8.0_222] {}
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708) ~[?:1.8.0_222] {}
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:1.8.0_222] {}
	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:566) ~[?:1.8.0_222] {}
	at net.minecraftforge.fml.ModLoader.gatherAndInitializeMods(ModLoader.java:172) ~[main/:?] {}
```

**Steps to Reproduce:**
I was trying to start Forge in the Forge development environment with the test mods.
However, the PistonEventTest is present in the test/java/../debug.block directory but not included in the test/resources/../mods.toml.

**Description of issue:**
Obviously this is an error. But Forge crashes when trying to log this fatal error and the game immediately closes.
The ModLoader creates an exception in #buildModContainerFromTOML, stores it for later and returns null. https://github.com/MinecraftForge/MinecraftForge/blob/88595dd26b87006ea4a360a75d48f2008f30b568/src/main/java/net/minecraftforge/fml/ModLoader.java#L232  
Therefore an element (not in the mods.toml) is added to the container list so the != here https://github.com/MinecraftForge/MinecraftForge/blob/88595dd26b87006ea4a360a75d48f2008f30b568/src/main/java/net/minecraftforge/fml/ModLoader.java#L212 is true. Inside that if statement the container list is serialized, which causes a NullPointer because of the added null element

**Comment by cpw** [comment]
OK

**Comment by maxanier** [comment]
Has been fixed in e784a7f606af103485e1cee7df03d958d1bf0614
